kernels/volk/volk_32fc_x2_s32fc_multiply_conjugate_32fc.h

```diff
@@ -0,0 +1,247 @@
+    const lv_32fc_t* b = bVector;
+    lv_32fc_t* c = cVector;
+
+    __m128 conjugator = _mm_setr_ps(0, -0.f, 0, -0.f);
+
+    // Set up constant scalar vector
+    sl = _mm_set_ps1(lv_creal(scalar));
+    sh = _mm_set_ps1(lv_cimag(scalar));
+
+    for(;number < halfPoints; number++){
+        x = _mm_load_ps((float*)b);
+        y = _mm_load_ps((float*)a);
+
+        x = _mm_xor_ps(x, conjugator);
+
+        tmp1 = _mm_mul_ps(x,sl);
+
+        x = _mm_shuffle_ps(x,x,0xB1);
+
+        tmp2 = _mm_mul_ps(x,sh);
+
+        z = _mm_addsub_ps(tmp1,tmp2);
+
+        z = _mm_add_ps(y, z);
+
+        _mm_store_ps((float*)c,z);
+
+        a += 2;
+        b += 2;
+        c += 2;
+    }
+
+    if((num_points % 2) != 0) {
+        *c = *a + lv_conj(*b) * scalar;
+    }
+}
+#endif /* LV_HAVE_SSE */
+
+#endif /* INCLUDED_volk_32fc_x2_s32fc_multiply_conjugate_32fc_a_H */
+
+
+#ifndef INCLUDED_volk_32fc_x2_s32fc_multiply_conjugate_32fc_u_H
+#define INCLUDED_volk_32fc_x2_s32fc_multiply_conjugate_32fc_u_H
+
+#include <inttypes.h>
+#include <stdio.h>
+#include <volk/volk_complex.h>
+#include <float.h>
+
+
+#ifdef LV_HAVE_NEON
+#include  <arm_neon.h>
+
+static inline void volk_32fc_x2_s32fc_multiply_conjugate_32fc_neon(lv_32fc_t* cVector, const lv_32fc_t* aVector, const lv_32fc_t* bVector, const lv_32fc_t scalar, unsigned int num_points){
+    const lv_32fc_t* bPtr = bVector;
+    const lv_32fc_t* aPtr = aVector;
+    lv_32fc_t* cPtr = cVector;
+    unsigned int number = num_points;
+    unsigned int quarter_points = num_points / 4;
+
+    float32x4x2_t a_val, b_val, c_val, scalar_val;
+    float32x4x2_t tmp_val;
+
+    scalar_val.val[0] = vld1q_dup_f32((const float*)&scalar);
+    scalar_val.val[1] = vld1q_dup_f32(((const float*)&scalar) + 1);
+
+    for(number = 0; number < quarter_points; ++number) {
+        a_val = vld2q_f32((float*)aPtr);
+        b_val = vld2q_f32((float*)bPtr);
+        b_val.val[1] = vnegq_f32(b_val.val[1]);
+        __VOLK_PREFETCH(aPtr + 8);
+        __VOLK_PREFETCH(bPtr + 8);
+
+        tmp_val.val[1] = vmulq_f32(b_val.val[1], scalar_val.val[0]);
+        tmp_val.val[0] = vmulq_f32(b_val.val[0], scalar_val.val[0]);
+
+        tmp_val.val[1] = vmlaq_f32(tmp_val.val[1], b_val.val[0], scalar_val.val[1]);
+        tmp_val.val[0] = vmlsq_f32(tmp_val.val[0], b_val.val[1], scalar_val.val[1]);
+
+        c_val.val[1] = vaddq_f32(a_val.val[1], tmp_val.val[1]);
+        c_val.val[0] = vaddq_f32(a_val.val[0], tmp_val.val[0]);
+
+        vst2q_f32((float*)cPtr, c_val);
+
+        aPtr += 4;
+        bPtr += 4;
+        cPtr += 4;
+    }
+
+    for(number = quarter_points*4; number < num_points; number++){
+        *cPtr++ = (*aPtr++) + lv_conj(*bPtr++) * scalar;
+    }
+}
+#endif /* LV_HAVE_NEON */
+
+#ifdef LV_HAVE_GENERIC
+
+static inline void volk_32fc_x2_s32fc_multiply_conjugate_32fc_generic(lv_32fc_t* cVector, const lv_32fc_t* aVector, const lv_32fc_t* bVector, const lv_32fc_t scalar, unsigned int num_points){
```

Comment:
Can you move the generic kernel to the top? In order to understand a kernel, I always look at the generic impl first. It's helpful to have it as the first impl.

Reply:
Yes, I will move the generic kernel to the top of this file.